spec/std/yaml/to_yaml_spec.cr

```diff
@@ -0,0 +1,36 @@
+require "spec"
+require "yaml"
+
+describe "String.to_yaml" do
+  values = {
+    "", "NULL", "Null", "null", "~",
+    "true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES",
+    "false", "False", "FALSE", "off", "Off", "OFF", "n", "N", "no", "No", "NO",
+    ".inf", ".Inf", ".INF",
+    "-.inf", "-.Inf", "-.INF",
+    ".nan", ".NaN", ".NAN",
+    "1", "1.0", "-1", "0",
+  }
+
+  values.each do |value|
+    it "should quote the non-string scalars" do
+      [value].to_yaml.should eq %(---\n- "#{value}"\n)
+    end
+  end
+end
+
+{% for type in %w(Float32 Float64) %}
```

Comment:
Should this be a non-macro style to match the way it was done in from_yaml_spec.cr  ?

Reply:
No, we pull out constants here. so it doesn't work the same way i.e. `{{type.id}}::NAN`